s/core.go

```diff
@@ -14,7 +15,23 @@ import (
 var rootSupervisorName = ""
 
 ////////////////////////////////////////////////////////////////////////////////
-// Supervisor (dynamic tree) functionality
+// Private API
+
```

Comment:
100% personal preference: Given that this is a public methods file, with some minor supporting functions, I often prefer to put the private cruft at the end and the API at the front. That way I see the API right away. Having the private functions hidden away at the bottom (specially if they have good names, as you do), is fine. Just a thought.

Reply:
Heh, for some reason I do it the opposite way (read files from bottom to top seems to be easier for me for some reason). Acknowledging this is a weird quirk of mine, I'm going to follow your advise and move them as you suggest.